pages/maps.tsx

```diff
@@ -398,57 +405,92 @@ const MapsGeneratorPage = () => {
             )}
           </FormControl>
         </>
-      ) : (
-        <Grid
-          mb={8}
-          templateColumns="repeat(4, 1fr)"
-          rowGap={4}
-          columnGap={4}
-          display={["none", null, "grid"]}
-        >
-          <Box textAlign="center">
-            <ModeImage mode="SZ" />
-          </Box>
-          <Box textAlign="center">
-            <ModeImage mode="TC" />
-          </Box>
-          <Box textAlign="center">
-            <ModeImage mode="RM" />
-          </Box>
-          <Box textAlign="center">
-            <ModeImage mode="CB" />
-          </Box>
-          <Box textAlign="center">
-            {Object.entries(stagesSelected).map(([stage, modes]) =>
-              modes.includes("SZ") ? (
-                <SubText key={stage}>{stage}</SubText>
-              ) : null
-            )}
-          </Box>
-          <Box textAlign="center">
-            {Object.entries(stagesSelected).map(([stage, modes]) =>
-              modes.includes("TC") ? (
-                <SubText key={stage}>{stage}</SubText>
-              ) : null
-            )}
-          </Box>
-          <Box textAlign="center">
-            {Object.entries(stagesSelected).map(([stage, modes]) =>
-              modes.includes("RM") ? (
-                <SubText key={stage}>{stage}</SubText>
-              ) : null
-            )}
-          </Box>
-          <Box textAlign="center">
-            {Object.entries(stagesSelected).map(([stage, modes]) =>
-              modes.includes("CB") ? (
-                <SubText key={stage}>{stage}</SubText>
-              ) : null
-            )}
-          </Box>
-        </Grid>
+      ) : (!isMobile &&
+        <>
```

Comment:
Could consider extracting the component here. The kind of nesting we have here is starting to make the component hard to read.